mbuild/tests/test_port.py

```diff
@@ -52,6 +52,20 @@ def test_port_direction(self):
         port.rotate(np.pi, [1, 0, 0])
         assert(np.allclose([0, -1, 0], port.direction, atol=1e-15))
 
+    def test_port_separation(self, ethane):
+        port = mb.Port(anchor=ethane, separation=0.7)
+        assert(np.allclose(0.7, port.separation, atol=1e-15))
+
+    def test_update_separation(self, ethane):
+        port = mb.Port(anchor=ethane, separation=0.7)
+        port.update_separation(separation=0.9)
+        assert(np.allclose(0.9, port.separation, atol=1e-15))
+
+    def test_update_orientaiton(self, ch2):
+        port = ch2['up']
+        port.update_orientation(orientation=(1,0,0))
+        assert(np.allclose([-1,0,0], port.direction, atol=1e-15))
+
```

Comment:
Ok one last test case and then I am happy with this PR. Lets include a test where we check that `None` is being returned and the proper warning message is being raised when a port is not anchored to another particle.

Reply:
Ok, the unit tests now makes sure that `None` is returned by `separation()` if there is no anchor particle.  Also, the 3 cases that should return warnings are tested as well.

I've never written a test for a warning message before, so I just used the simplest method I could find. It's basically only checking that a warning did occur and nothing about the type or content of the warning.